packages/loot-core/src/server/accounts/rules.js

```diff
@@ -146,6 +146,28 @@ let CONDITION_TYPES = {
         );
       }
 
+      if (op === 'matches') {
+        assert(
+          typeof value === 'string' && value.length > 0,
+          'no-empty-string',
+          `matches must have non-empty string (field: ${fieldName})`,
+        );
+        var isValidRegexp = true;
+        try {
+          new RegExp(value, 'ui');
+        } catch (e) {
+          isValidRegexp = false;
```

Comment:
Could the error message be captured too so that users can debug why their regex is invalid?